edx_notifications/lib/consumer.py

```diff
@@ -0,0 +1,121 @@
+"""
+All in-proc API endpoints for acting as a Notifications Consumer
+NOTE that we can only query for notifications that are "durable", i.e.
+persisted in our database
+"""
+
+import pytz
+from datetime import datetime
+
+from edx_notifications.stores.store import notification_store
+
+from edx_notifications.exceptions import (
+    ItemIntegrityError
+)
+
+
+def get_notifications_count_for_user(user_id, filters=None):
+    """
+    Returns the number of notifications this user has
+
+    ARGS:
+        user_id: The id of the user to query
+        filters: A dict containing the following optional keys
+            - 'namespace': what namespace to search (defuault None)
+            - 'read': Whether to return read notifications (default True)
+            - 'unread': Whether to return unread notifications (default True)
+            - 'type_name': what msg_types to count
+
+    NOTE: If the 'msg_id' filter is passed in, a ValueError will be raised as
+    this is not a valid combination
+    """
+
+    # make sure user_id is an integer
+    if not isinstance(user_id, int):
+        raise TypeError("user_id must be an integer")
+
+    return notification_store().get_num_notifications_for_user(
+        user_id,
+        filters=filters,
+    )
+
+
+def get_notifications_for_user(user_id, filters=None, options=None):
+    """
+    Returns a list of NotificationUserMap for the passed in user. The list will be
+    ordered by most recent first
+
+    ARGS:
+        - user_id: The id of the user
+        - filters: a dict containing
+            - msg_id: a particular notification id to retrieve
+                ( if this is used in conjuction with other filters, it may
+                  result in an empty set being returned )
+            - namespace: what namespace to search (defuault None)
+            - read: Whether to return read notifications (default True)
+            - unread: Whether to return unread notifications (default True)
+            - type_name: which type to return
+        - options: a dict containing some optional parameters
+            - limit: max number to return (up to some system defined max)
+            - offset: offset into the list, to implement paging
+    """
+
+    # make sure user_id is an integer
+    if not isinstance(user_id, int):
+        raise TypeError("user_id must be an integer")
```

Comment:
You should make sure you call `contracts.disable_all()` when this runs as an independent process, so that there's no overhead to the contracts calls.